features/collections/collection_anonymity.feature

```diff
@@ -375,3 +375,81 @@ Feature: Collection
     # This is not the desired behavior (AO3-5556), but we want to make sure it doesn't get broken worse
     Then I should see "Anonymous Collection"
       And I should see "Anonymous [creator]"
+
+  Scenario: When an anonymous collection is deleted, works in the collection stop being anonymous.
+    Given I have an anonymous collection "Anonymous Collection"
+      And I am logged in as "creator"
+      And I post the work "Secret Work" to the collection "Anonymous Collection"
+
+    When I go to my works page
+    Then I should not see "Secret Work"
+
+    When I am logged in as the owner of "Anonymous Collection"
+      And I go to "Anonymous Collection" collection edit page
+      And I follow "Delete Collection"
+      And I press "Yes, Delete Collection"
+      And I go to creator's works page
+    Then I should see "Secret Work"
+
+  Scenario: When an unrevealed collection is deleted, works in the collection stop being unrevealed.
+    Given I have a hidden collection "Hidden Collection"
+      And I am logged in as "creator"
+      And I post the work "Secret Work" to the collection "Hidden Collection"
+
+    When I view the work "Secret Work"
+    Then I should see "This work is part of an ongoing challenge and will be revealed soon!"
+
+    When I am logged in as the owner of "Hidden Collection"
+      And I go to "Hidden Collection" collection edit page
+      And I follow "Delete Collection"
+      And I press "Yes, Delete Collection"
+      And I view the work "Secret Work"
+    Then I should not see "This work is part of an ongoing challenge and will be revealed soon!"
```

Comment:
We could use `Then I should see "Secret Work"` so the check will work even if we change the unrevealed work notice.

Reply:
I'll have to add an `And I am logged out` step for that, since in the tests I wrote the viewer is either the work's creator or the collection's owner, both of whom can always see the work's title even when it's hidden. But that's a good idea, thanks!